compiler/plugins/target/AMD-AIE/iree-amd-aie/IR/AMDAIEOps.td

```diff
@@ -586,6 +586,21 @@ def AMDAIE_NpuDmaCpyNdOp: AMDAIE_Op<"npu.dma_cpy_nd", [
   let hasCanonicalizer = 1;
 }
 
+def AMDAIE_NpuDmaPlaceHolderOp : AMDAIE_Op<"npu.dma_placeholder"> {
+  let summary = "Represents a placeholder for a DMA operation.";
+  let description = [{
+    This operation acts as a placeholder user for those `amdaie.connection` operations
+    that represent control flows, to prevent them from being dead-code eliminated.
+    This operation does not have any side effects on control code size.
```

Comment:
I would write this a bit more generally as there could be potential other uses for this placeholder in the future.
```suggestion
    This operation acts as a placeholder user for `amdaie.connection` operations to prevent
    them from being dead-code eliminated. This is used for control flow connections that are
    inserted before control packets are generated because they need to be taken into account
    together with data connections for routing. This operation does not have any side effects
    on control code size.
```